huaweicloud/services/cce/resource_huaweicloud_cce_partition.go

```diff
@@ -0,0 +1,222 @@
+			Name: d.Get("availability_zone").(string),
+		},
+		Spec: partitions.Spec{
+			Category:          d.Get("category").(string),
+			PublicBorderGroup: d.Get("availability_zone").(string),
+			HostNetwork:       resourcePartitionSubnetID(d),
+			ContainerNetwork:  resourceContainerSubnetIDs(d),
+		},
+	}
+
+	log.Printf("[DEBUG] Create Options: %#v", createOpts)
+
+	s, err := partitions.Create(cceClient, clusterid, createOpts).Extract()
+	if err != nil {
+		return diag.Errorf("Error creating Partition: %s", err)
+	}
+	d.SetId(s.Metadata.Name)
+
+	return resourcePartitionRead(ctx, d, meta)
+}
+
+func resourcePartitionRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	cceClient, err := cfg.CceV3Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("Error creating CCE client: %s", err)
+	}
+	clusterid := d.Get("cluster_id").(string)
+	s, err := partitions.Get(cceClient, clusterid, d.Id()).Extract()
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "Error retrieving CCE Partition")
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("region", cfg.GetRegion(d)),
+		d.Set("name", s.Metadata.Name),
+		d.Set("category", s.Spec.Category),
+		d.Set("availability_zone", s.Spec.PublicBorderGroup),
+		d.Set("partition_subnet_id", s.Spec.HostNetwork.SubnetID),
+	)
+
+	var containerNetworkIDs []string
+	for _, network := range s.Spec.ContainerNetwork {
+		containerNetworkIDs = append(containerNetworkIDs, network.SubnetID)
+	}
+	mErr = multierror.Append(mErr, d.Set("container_subnet_ids", containerNetworkIDs))
+
+	if err = mErr.ErrorOrNil(); err != nil {
+		return diag.Errorf("Error setting CCE Partition fields: %s", err)
+	}
+	return nil
+}
+
+func resourcePartitionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	cceClient, err := cfg.CceV3Client(region)
+	if err != nil {
+		return diag.Errorf("Error creating CCE client: %s", err)
+	}
+
+	if d.HasChange("container_subnet_ids") {
+		var updateOpts partitions.UpdateOpts
+		updateOpts.Metadata.ContainerNetwork = resourceContainerSubnetIDs(d)
+
+		clusterid := d.Get("cluster_id").(string)
+		_, err = partitions.Update(cceClient, clusterid, d.Id(), updateOpts).Extract()
+		if err != nil {
+			return diag.Errorf("Error updating cce partition: %s", err)
+		}
+	}
+
+	return resourceCCENodeV3Read(ctx, d, meta)
+}
+
+func resourcePartitionDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	cceClient, err := cfg.CceV3Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("Error creating CCE client: %s", err)
+	}
+
+	clusterid := d.Get("cluster_id").(string)
+	err = partitions.Delete(cceClient, clusterid, d.Id()).ExtractErr()
+	if err != nil {
+		return diag.Errorf("Error deleting CCE partition: %s", err)
+	}
+
+	d.SetId("")
```

Comment:
we can delete this line as the plugin-sdk will do it.

Reply:
ok